pkgs/by-name/ko/koboldcpp/package.nix

```diff
@@ -28,38 +29,53 @@
 
   metalSupport ? stdenv.isDarwin && stdenv.isAarch64,
 
+  # You can find a full list here: https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/
+  # For example if you're on an GTX 1080 that means you're using "Pascal" and you need to pass "6.0"
+  cudaCapabilities ? cudaPackages.cudaFlags.cudaCapabilities or [ ],
+
   # You can find list of x86_64 options here: https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
   # For ARM here: https://gcc.gnu.org/onlinedocs/gcc/ARM-Options.html
   # If you set "march" to "native", specify "mtune" as well; otherwise, it will be set to "generic". (credit to: https://lemire.me/blog/2018/07/25/it-is-more-complicated-than-i-thought-mtune-march-in-gcc/)
-  # For example, if you have an AMD Ryzen CPU, you will set "march" to "x86-64" and "mtune" to "znver2"
+  # For example, if you have an AMD Ryzen CPU, you will set "mtune" to "znver2"
   march ? "",
+  # Apple Silicon Chips (M1, M2, M3 and so on DO NOT set mtune)
   mtune ? "",
+  # Instead set mcpu
+  mcpu ? "",
 }:
 
 let
   makeBool = option: bool: (if bool then "${option}=1" else "");
 
+  makeWrapperArgs = lib.optionalString config.cudaSupport ''
+    --prefix LD_LIBRARY_PATH: "${lib.makeLibraryPath [ addDriverRunpath.driverLink ]}"
+  '';
+
+  containsValidCapability = lib.any (
+    x: x == builtins.head cudaCapabilities
+  ) cudaPackages.cudaFlags.cudaCapabilities;
```

Comment:
I think it's ok to leave this logic out; the default you've set after `cudaCapabilities ? ...` is guaranteed to be correct, and if the package is called with different args it must be that the user supplied them in an override; the user probably knows better what they want

Reply:
Yeah that's true, I will remove this